src/utils.jl

```diff
@@ -0,0 +1,184 @@
+# todo define methods and remove many of the function below to use multiple dispatch
+
+immutable Double2
+x::Float64
```

Comment:
Probably don't want to add it as a dependency, but feel free to copy what you need.